src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String data = readFile(fromFileName);
+            String report = generateReport(data);
+            writeToFile(toFileName, report);
+        } catch (IOException e) {
+            System.out.println("An error occurred while reading/writing the file: "
+                    + e.getMessage());
+        }
```

Comment:
better move catch to read/write methods, rethrow runtime in catch with message, filename